Documentation/configuration/api-rate-limiting.rst

```diff
@@ -0,0 +1,135 @@
+plugin is invoked when a new workload is scheduled onto the node which in turn
+makes an API call to the Cilium agent to allocate an IP address and create the
+Cilium endpoint. Another example is loading of network policy or services
+definitions where changes of these definitions will create an event which will
+notify the Cilium agent that a modification is required.
+
+Due to being event-driven, the amount of work performed by the Cilium agent
+highly depends on the rate of external events it receives. In order to
+constrain the resources that the Cilium agent consumes, it can be helpful to
+restrict the rate and allowed parallel executions of API calls.
+
+Default Rate Limits
+===================
+
+The following API calls are currently subject to rate limiting:
+
+========================== ====== ===== ============= ============ ================= =========== ===============================
+API Call                   Limit  Burst Max Parallel  Min Parallel Max Wait Duration Auto Adjust Estimated Processing Duration
+========================== ====== ===== ============= ============ ================= =========== ===============================
+``PUT /endpoint/{id}``     0.5/s  4     4                          15s               True        2s
+``DELETE /endpoint/{id}``               4             4                              True        200ms
+``GET /endpoint/{id}/*``   4/s    4     4             2            10s               True        200ms
+``PATCH /endpoint/{id}*``  0.5/s  4     4                          15s               True        1s
+``GET /endpoint``          1/s    4     2             2                              True        300ms
+========================== ====== ===== ============= ============ ================= =========== ===============================
+
+Configuration
+=============
+
+The ``api-rate-limit`` option can be used to overwrite individual settings of the
+default configuration:
+
+.. code::
+
+   --api-rate-limit endpoint-create=rate-limit:2/s,rate-burst:4
+
+API call to Configuration mapping
+---------------------------------
+
+========================== ====================
+API Call                   Config Name
+========================== ====================
+``PUT /endpoint/{id}``     ``endpoint-create``
+``DELETE /endpoint/{id}``  ``endpoint-delete``
+``GET /endpoint/{id}/*``   ``endpoint-get``
+``PATCH /endpoint/{id}*``  ``endponit-patch``
+``GET /endpoint``          ``endpoint-list``
+========================== ==================== 
+
+Configuration Parameters
+------------------------
+
+================================= ========= ==========================================================================
+Configuration Key                 Example   Description
+================================= ========= ==========================================================================
+``rate-limit``                    ``5/m``   Allowed requests per time unit.
```

Comment:
I don't see us calling out what the accepted time unit abbreviations are. I think it would help the user if we have a small section to say:

```
"m": minutes
"s": seconds
"ms": milliseconds
...
```